openpype/plugins/publish/collect_input_representations_to_versions.py

```diff
@@ -22,26 +22,30 @@ def process(self, context):
         # Query all version ids for representation ids from the database once
         representations = set()
         for instance in context:
-            inst_repre = instance.data.get("inputRepresentations", [])
-            representations.update(inst_repre)
+            inst_repre = instance.data.get("inputRepresentations")
+            if inst_repre:
+                representations.update(inst_repre)
 
         representations_docs = get_representations(
             project_name=context.data["projectEntity"]["name"],
             representation_ids=representations,
             fields=["_id", "parent"])
 
         representation_id_to_version_id = {
-            repre["_id"]: repre["parent"] for repre in representations_docs
+            str(repre["_id"]): repre["parent"]
+            for repre in representations_docs
         }
 
         for instance in context:
-            inst_repre = instance.data.get("inputRepresentations", [])
+            inst_repre = instance.data.get("inputRepresentations")
             if not inst_repre:
                 continue
 
-            input_versions = instance.data.get("inputVersions", [])
+            if "inputVersions" not in instance.data:
+                instance.data["inputVersions"] = []
+            input_versions = instance.data["inputVersions"]
```

Comment:
This would be better written as:
```suggestion
input_versions = instance.data.setdefault("inputVersions", [])
```

Also since we're now assigning it beforehand to `instance.data["inputVersions"]` I feel the `instance.data["inputVersions"] = input_versions` afterwards is redundant since we're basically assigning the same list again.